platform/linux/config.cmake

```diff
@@ -15,19 +15,31 @@ mason_use(benchmark VERSION 1.0.0)
 include(cmake/loop-uv.cmake)
 
 macro(mbgl_platform_core)
-    if (WITH_OSMESA)
+    if(WITH_OSMESA)
         target_sources(mbgl-core
             PRIVATE platform/default/headless_backend_osmesa.cpp
             PRIVATE platform/default/headless_display.cpp
         )
         target_add_mason_package(mbgl-core PUBLIC mesa)
+    elseif(WITH_EGL)
+        target_sources(mbgl-core
+            PRIVATE platform/linux/src/headless_backend_egl.cpp
+            PRIVATE platform/linux/src/headless_display_egl.cpp
+        )
+        # XXX: Provide surface-less EGL mesa for CI builds.
```

Comment:
Yeah - https://github.com/mapbox/mapbox-gl-native/issues/7020
I'll add that link to the comment 👍